packages/react-devtools-core/src/standalone.js

```diff
@@ -200,6 +200,14 @@ var DevtoolsUI = {
     return DevtoolsUI;
   },
 
+  setBrowserName(name) {
+    config.browserName = name;
+    if (panel) {
+      panel.props.browserName = name;
```

Comment:
One way to do this might be to move `browserName` to the `Store` and then use a similar approach as with `setDefaultThemeName`:
```diff
diff --git a/frontend/Panel.js b/frontend/Panel.js
index 34ed519..e164619 100644
--- a/frontend/Panel.js
+++ b/frontend/Panel.js
@@ -100,7 +100,7 @@ class Panel extends React.Component {
 
   getChildContext(): Object {
     return {
-      browserName: this.props.browserName || '',
+      browserName: this._store && this._store.browserName || '',
       defaultThemeName: this._themeStore && this._themeStore.defaultThemeName || '',
       showHiddenThemes: !!this.props.showHiddenThemes,
       showInspectButton: this.props.showInspectButton !== false,
@@ -221,7 +221,11 @@ class Panel extends React.Component {
       this._bridge = new Bridge(wall);
 
       this._themeStore = new ThemeStore(this.state.themeName);
-      this._store = new Store(this._bridge, this._themeStore);
+      this._store = new Store(
+        this._bridge,
+        this._themeStore,
+        this.props.browserName,
+      );
       
       var refresh = () => this.forceUpdate();
       this.plugins = [
diff --git a/frontend/PreferencesPanel.js b/frontend/PreferencesPanel.js
index d5473bd..8fc5d2a 100644
--- a/frontend/PreferencesPanel.js
+++ b/frontend/PreferencesPanel.js
@@ -24,13 +24,13 @@ class PreferencesPanel extends React.Component {
   _selectRef: any;
 
   context: {
-    browserName: string,
     showHiddenThemes: boolean,
     theme: Theme,
     themeName: string,
     themes: { [key: string]: Theme },
   };
   props: {
+    browserName: string,
     changeTheme: (themeName: string) => void,
     hasCustomTheme: boolean,
     hide: () => void,
@@ -61,8 +61,8 @@ class PreferencesPanel extends React.Component {
   }
 
   render() {
-    const {browserName, showHiddenThemes, theme, themeName, themes} = this.context;
-    const {hasCustomTheme, hide, open} = this.props;
+    const {showHiddenThemes, theme, themeName, themes} = this.context;
+    const {browserName, hasCustomTheme, hide, open} = this.props;
     const {editMode} = this.state;
 
     if (!open) {
@@ -161,13 +161,13 @@ class PreferencesPanel extends React.Component {
 }
 
 PreferencesPanel.contextTypes = {
-  browserName: React.PropTypes.string.isRequired,
   showHiddenThemes: React.PropTypes.bool.isRequired,
   theme: React.PropTypes.object.isRequired,
   themeName: React.PropTypes.string.isRequired,
   themes: React.PropTypes.object.isRequired,
 };
 PreferencesPanel.propTypes = {
+  browserName: React.PropTypes.string.isRequired,
   changeTheme: React.PropTypes.func,
   hide: React.PropTypes.func,
   open: React.PropTypes.bool,
@@ -184,10 +184,11 @@ const blockClick = event => event.stopPropagation();
 
 const WrappedPreferencesPanel = decorate({
   listeners() {
-    return ['preferencesPanelShown'];
+    return ['browserName', 'preferencesPanelShown'];
   },
   props(store, props) {
     return {
+      browserName: store.browserName || '',
       changeTheme: themeName => store.changeTheme(themeName),
       hasCustomTheme: !!store.themeStore.customTheme,
       hide: () => store.hidePreferencesPanel(),
diff --git a/frontend/Store.js b/frontend/Store.js
index 67b4465..6c9ff70 100644
--- a/frontend/Store.js
+++ b/frontend/Store.js
@@ -89,6 +89,7 @@ class Store extends EventEmitter {
   _eventTimer: ?number;
 
   // Public state
+  browserName: ?string;
   isInspectEnabled: boolean;
   traceupdatesState: ?ControlState;
   colorizerState: ?ControlState;
@@ -112,7 +113,11 @@ class Store extends EventEmitter {
     rnStyleMeasure?: boolean,
   };
 
-  constructor(bridge: Bridge, themeStore: ThemeStore) {
+  constructor(
+    bridge: Bridge,
+    themeStore: ThemeStore,
+    browserName: ?string,
+  ) {
     super();
 
     this._nodes = new Map();
@@ -121,6 +126,7 @@ class Store extends EventEmitter {
     this._bridge = bridge;
 
     // Public state
+    this.browserName = browserName;
     this.isInspectEnabled = false;
     this.roots = new List();
     this.contextMenu = null;
@@ -345,6 +351,11 @@ class Store extends EventEmitter {
     this.emit('theme');
   }
 
+  setBrowserName(browserName: ?string) {
+    this.browserName = browserName;
+    this.emit('browserName');
+  }
+
   setDefaultThemeName(defaultThemeName: ?string) {
     this.themeStore.setDefaultThemeName(defaultThemeName);
   }
diff --git a/packages/react-devtools-core/src/standalone.js b/packages/react-devtools-core/src/standalone.js
index 8b28055..231a109 100644
--- a/packages/react-devtools-core/src/standalone.js
+++ b/packages/react-devtools-core/src/standalone.js
@@ -199,10 +199,12 @@ var DevtoolsUI = {
     return DevtoolsUI;
   },
 
-  setBrowserName(name) {
-    config.browserName = name;
+  setBrowserName(browserName) {
+    config.browserName = browserName;
     if (panel) {
-      panel.props.browserName = name;
+      var {store} = panel.getChildContext();
+      // Change browserName if panel already mounted
+      store.setBrowserName(browserName);
     }
     return DevtoolsUI;
   },
```